testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestHiveTransactionalTable.java

```diff
@@ -1289,6 +1289,410 @@ public void testDeleteWholePartition()
+            });
+        });
+    }
+
+    @Test(groups = HIVE_TRANSACTIONAL, timeOut = TEST_TIMEOUT, dataProvider = "partitionedAndBucketedProvider")
+    public void testMergeUpdateWithVariousLayouts(boolean partitioned, String bucketing)
+    {
+        log.info("In testMergeUpdateWithVariousLayouts, partitioned %s, bucketing %s", partitioned, bucketing);
+        BucketingType bucketingType = bucketing.isEmpty() ? NONE : BUCKETED_V2;
+        withTemporaryTable("merge_with_various_formats", true, true, bucketingType, targetTable -> {
+            StringBuilder builder = new StringBuilder();
+            builder.append("CREATE TABLE ")
+                    .append(targetTable)
+                    .append("(customer STRING");
+            builder.append(partitioned ? ") PARTITIONED BY (" : ", ")
+                    .append("purchase STRING) ");
+            if (!bucketing.isEmpty()) {
+                builder.append(bucketing);
+            }
+            builder.append(" STORED AS ORC TBLPROPERTIES ('transactional' = 'true')");
+            onHive().executeQuery(builder.toString());
+
+            log.info("About to insert");
+            query(format("INSERT INTO %s (customer, purchase) VALUES ('Dave', 'dates'), ('Lou', 'limes'), ('Carol', 'candles')", targetTable));
+            verifySelectForPrestoAndHive("SELECT * FROM " + targetTable, "TRUE", row("Dave", "dates"), row("Lou", "limes"), row("Carol", "candles"));
+
+            withTemporaryTable("merge_simple_source", true, true, NONE, sourceTable -> {
+                query(format("CREATE TABLE %s (customer VARCHAR, purchase VARCHAR) WITH (transactional = true)", sourceTable));
+
+                log.info("Inserting into source");
+                query(format("INSERT INTO %s (customer, purchase) VALUES ('Craig', 'candles'), ('Len', 'limes'), ('Joe', 'jellybeans')", sourceTable));
+
+                String sql = format("MERGE INTO %s t USING %s s ON (t.purchase = s.purchase)", targetTable, sourceTable) +
+                        "    WHEN MATCHED AND s.purchase = 'limes' THEN DELETE" +
+                        "    WHEN MATCHED THEN UPDATE SET customer = CONCAT(t.customer, '_', s.customer)" +
+                        "    WHEN NOT MATCHED THEN INSERT (customer, purchase) VALUES(s.customer, s.purchase)";
+
+                log.info("About to merge, target table %s, source table %s", targetTable, sourceTable);
+                query(sql);
+
+                log.info("Verifying MERGE");
+                verifySelectForPrestoAndHive("SELECT * FROM " + targetTable, "TRUE", row("Dave", "dates"), row("Carol_Craig", "candles"), row("Joe", "jellybeans"));
+            });
+        });
+    }
+
+    @Test(groups = HIVE_TRANSACTIONAL, timeOut = TEST_TIMEOUT)
+    public void testMergeUnBucketedUnPartitionedFailure()
```

Comment:
What Failure is here?

Reply:
There is a log line at the end that implies the problem:

                             `log.info("Verifying MERGE on Hive fails - - and it shouldn't");`

But it needs a real comment.  Here is the comment I added:

```
    /**
     * This test demonstrates a failure of Hive to verify the result of a MERGE operation,
     * specifically, Hive fails to recognize the delete_delta file written by the MERGE.  I
     * captured the HDFS delta and delete_delta files and verified that they are correct.
     * I used Wireshark to capture the traffic between Trino and the Hive metastore during
     * the MERGE, and it was all as expected.  I tried to vary the test to understand the
     * issue, but almost any change I made to the test caused Hive to correctly verify the
     * MERGE.
     * TODO: Determine what is causing the Hive verification failure 
     */
```

Reply:
Thanks for pointing this out. This looks like a bug somewhere and requires an action item. Can you please make sure this gets properly addressed?